From 3d6d52db0b81a70a6fb0ff509cb31224c4857788 Mon Sep 17 00:00:00 2001
From: Emmanuel Blot <emmanuel.blot@free.fr>
Date: Fri, 7 Aug 2026 17:23:08 +0200
Subject: [PATCH] hw/arm: bletchley: populate the v1.5 FRU EEPROMs

Add the v1.5 image set, so "board-revision" now yields two distinct
FRU layouts.

Two interposers were missing from the source chassis; their records
are reissued from a populated sled so both revisions describe the same
topology. Content is sanitized as before.

Signed-off-by: Emmanuel Blot <emmanuel.blot@free.fr>
---
 hw/arm/aspeed_ast2600_bletchley.c | 287 +++++++++++++++++++++++++++++-
 1 file changed, 286 insertions(+), 1 deletion(-)

diff --git a/hw/arm/aspeed_ast2600_bletchley.c b/hw/arm/aspeed_ast2600_bletchley.c
index c3f7dfb27b..6439859504 100644
--- a/hw/arm/aspeed_ast2600_bletchley.c
+++ b/hw/arm/aspeed_ast2600_bletchley.c
@@ -332,6 +332,290 @@ static const BletchleyFruSet bletchley_fru_v1_0 = {
     .cc  = { fru_v1_0_7_0054, sizeof(fru_v1_0_7_0054) },
 };
 
+/* Bletchley v1.5 */
+
+/*
+ * FRU: i2c0 @0x54 (sled 1 interposer)
+ *   Board mfg:       Quanta
+ *   Board product:   Bletchley Interposer-Class 1
+ *   Board part:      0AA00AA0001
+ *   Board serial:    AAA0A00000011
+ *   Board mfg date:  2026-01-01T00:00:00Z
+ *   Board file ID:   FRU Ver 0.04
+ *   Product mfg:     Quanta
+ *   Product name:    Bletchley MP
+ *   Product part:    0AA00AA00A2
+ *   Product version: MP
+ *   Product serial:  AAAA0000000012
+ *   Product asset:   QEMU
+ */
+static const uint8_t fru_v1_5_0_0054[] = {
+    0x01, 0x00, 0x00, 0x01, 0x0c, 0x00, 0x00, 0xf2, 0x01, 0x0b, 0x19, 0xc0,
+    0xc6, 0xf0, 0xc6, 0x51, 0x75, 0x61, 0x6e, 0x74, 0x61, 0xdc, 0x42, 0x6c,
+    0x65, 0x74, 0x63, 0x68, 0x6c, 0x65, 0x79, 0x20, 0x49, 0x6e, 0x74, 0x65,
+    0x72, 0x70, 0x6f, 0x73, 0x65, 0x72, 0x2d, 0x43, 0x6c, 0x61, 0x73, 0x73,
+    0x20, 0x31, 0xcd, 0x41, 0x41, 0x41, 0x30, 0x41, 0x30, 0x30, 0x30, 0x30,
+    0x30, 0x30, 0x31, 0x31, 0xcb, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41,
+    0x30, 0x30, 0x30, 0x31, 0xcc, 0x46, 0x52, 0x55, 0x20, 0x56, 0x65, 0x72,
+    0x20, 0x30, 0x2e, 0x30, 0x34, 0xc1, 0x00, 0x00, 0x00, 0x00, 0x00, 0xb2,
+    0x01, 0x08, 0x19, 0xc6, 0x51, 0x75, 0x61, 0x6e, 0x74, 0x61, 0xcc, 0x42,
+    0x6c, 0x65, 0x74, 0x63, 0x68, 0x6c, 0x65, 0x79, 0x20, 0x4d, 0x50, 0xcb,
+    0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x32, 0xc2,
+    0x4d, 0x50, 0xce, 0x41, 0x41, 0x41, 0x41, 0x30, 0x30, 0x30, 0x30, 0x30,
+    0x30, 0x30, 0x30, 0x31, 0x32, 0xc4, 0x51, 0x45, 0x4d, 0x55, 0x00, 0xc1,
+    0x00, 0x00, 0x00, 0x86,
+};
+
+/*
+ * FRU: i2c1 @0x54 (sled 2 interposer)
+ *   Board mfg:       Quanta
+ *   Board product:   Bletchley Interposer-Class 1
+ *   Board part:      0AA00AA0001
+ *   Board serial:    AAA0A00000013
+ *   Board mfg date:  2026-01-01T00:00:00Z
+ *   Board file ID:   FRU Ver 0.04
+ *   Product mfg:     Quanta
+ *   Product name:    Bletchley MP
+ *   Product part:    0AA00AA00A2
+ *   Product version: MP
+ *   Product serial:  AAAA0000000012
+ *   Product asset:   QEMU
+ */
+static const uint8_t fru_v1_5_1_0054[] = {
+    0x01, 0x00, 0x00, 0x01, 0x0c, 0x00, 0x00, 0xf2, 0x01, 0x0b, 0x19, 0xc0,
+    0xc6, 0xf0, 0xc6, 0x51, 0x75, 0x61, 0x6e, 0x74, 0x61, 0xdc, 0x42, 0x6c,
+    0x65, 0x74, 0x63, 0x68, 0x6c, 0x65, 0x79, 0x20, 0x49, 0x6e, 0x74, 0x65,
+    0x72, 0x70, 0x6f, 0x73, 0x65, 0x72, 0x2d, 0x43, 0x6c, 0x61, 0x73, 0x73,
+    0x20, 0x31, 0xcd, 0x41, 0x41, 0x41, 0x30, 0x41, 0x30, 0x30, 0x30, 0x30,
+    0x30, 0x30, 0x31, 0x33, 0xcb, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41,
+    0x30, 0x30, 0x30, 0x31, 0xcc, 0x46, 0x52, 0x55, 0x20, 0x56, 0x65, 0x72,
+    0x20, 0x30, 0x2e, 0x30, 0x34, 0xc1, 0x00, 0x00, 0x00, 0x00, 0x00, 0xb0,
+    0x01, 0x08, 0x19, 0xc6, 0x51, 0x75, 0x61, 0x6e, 0x74, 0x61, 0xcc, 0x42,
+    0x6c, 0x65, 0x74, 0x63, 0x68, 0x6c, 0x65, 0x79, 0x20, 0x4d, 0x50, 0xcb,
+    0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x32, 0xc2,
+    0x4d, 0x50, 0xce, 0x41, 0x41, 0x41, 0x41, 0x30, 0x30, 0x30, 0x30, 0x30,
+    0x30, 0x30, 0x30, 0x31, 0x32, 0xc4, 0x51, 0x45, 0x4d, 0x55, 0x00, 0xc1,
+    0x00, 0x00, 0x00, 0x86,
+};
+
+/*
+ * FRU: i2c2 @0x54 (sled 3 interposer)
+ *   (sled absent from the source chassis, serial reissued)
+ *   Board mfg:       Quanta
+ *   Board product:   Bletchley Interposer-Class 1
+ *   Board part:      0AA00AA0001
+ *   Board serial:    AAA0A00000019
+ *   Board mfg date:  2026-01-01T00:00:00Z
+ *   Board file ID:   FRU Ver 0.04
+ *   Product mfg:     Quanta
+ *   Product name:    Bletchley MP
+ *   Product part:    0AA00AA00A2
+ *   Product version: MP
+ *   Product serial:  AAAA0000000012
+ *   Product asset:   QEMU
+ */
+static const uint8_t fru_v1_5_2_0054[] = {
+    0x01, 0x00, 0x00, 0x01, 0x0c, 0x00, 0x00, 0xf2, 0x01, 0x0b, 0x19, 0xc0,
+    0xc6, 0xf0, 0xc6, 0x51, 0x75, 0x61, 0x6e, 0x74, 0x61, 0xdc, 0x42, 0x6c,
+    0x65, 0x74, 0x63, 0x68, 0x6c, 0x65, 0x79, 0x20, 0x49, 0x6e, 0x74, 0x65,
+    0x72, 0x70, 0x6f, 0x73, 0x65, 0x72, 0x2d, 0x43, 0x6c, 0x61, 0x73, 0x73,
+    0x20, 0x31, 0xcd, 0x41, 0x41, 0x41, 0x30, 0x41, 0x30, 0x30, 0x30, 0x30,
+    0x30, 0x30, 0x31, 0x39, 0xcb, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41,
+    0x30, 0x30, 0x30, 0x31, 0xcc, 0x46, 0x52, 0x55, 0x20, 0x56, 0x65, 0x72,
+    0x20, 0x30, 0x2e, 0x30, 0x34, 0xc1, 0x00, 0x00, 0x00, 0x00, 0x00, 0xaa,
+    0x01, 0x08, 0x19, 0xc6, 0x51, 0x75, 0x61, 0x6e, 0x74, 0x61, 0xcc, 0x42,
+    0x6c, 0x65, 0x74, 0x63, 0x68, 0x6c, 0x65, 0x79, 0x20, 0x4d, 0x50, 0xcb,
+    0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x32, 0xc2,
+    0x4d, 0x50, 0xce, 0x41, 0x41, 0x41, 0x41, 0x30, 0x30, 0x30, 0x30, 0x30,
+    0x30, 0x30, 0x30, 0x31, 0x32, 0xc4, 0x51, 0x45, 0x4d, 0x55, 0x00, 0xc1,
+    0x00, 0x00, 0x00, 0x86,
+};
+
+/*
+ * FRU: i2c3 @0x54 (sled 4 interposer)
+ *   Board mfg:       Quanta
+ *   Board product:   Bletchley Interposer-Class 1
+ *   Board part:      0AA00AA0001
+ *   Board serial:    AAA0A00000014
+ *   Board mfg date:  2026-01-01T00:00:00Z
+ *   Board file ID:   FRU Ver 0.04
+ *   Product mfg:     Quanta
+ *   Product name:    Bletchley MP
+ *   Product part:    0AA00AA00A2
+ *   Product version: MP
+ *   Product serial:  AAAA0000000012
+ *   Product asset:   QEMU
+ */
+static const uint8_t fru_v1_5_3_0054[] = {
+    0x01, 0x00, 0x00, 0x01, 0x0c, 0x00, 0x00, 0xf2, 0x01, 0x0b, 0x19, 0xc0,
+    0xc6, 0xf0, 0xc6, 0x51, 0x75, 0x61, 0x6e, 0x74, 0x61, 0xdc, 0x42, 0x6c,
+    0x65, 0x74, 0x63, 0x68, 0x6c, 0x65, 0x79, 0x20, 0x49, 0x6e, 0x74, 0x65,
+    0x72, 0x70, 0x6f, 0x73, 0x65, 0x72, 0x2d, 0x43, 0x6c, 0x61, 0x73, 0x73,
+    0x20, 0x31, 0xcd, 0x41, 0x41, 0x41, 0x30, 0x41, 0x30, 0x30, 0x30, 0x30,
+    0x30, 0x30, 0x31, 0x34, 0xcb, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41,
+    0x30, 0x30, 0x30, 0x31, 0xcc, 0x46, 0x52, 0x55, 0x20, 0x56, 0x65, 0x72,
+    0x20, 0x30, 0x2e, 0x30, 0x34, 0xc1, 0x00, 0x00, 0x00, 0x00, 0x00, 0xaf,
+    0x01, 0x08, 0x19, 0xc6, 0x51, 0x75, 0x61, 0x6e, 0x74, 0x61, 0xcc, 0x42,
+    0x6c, 0x65, 0x74, 0x63, 0x68, 0x6c, 0x65, 0x79, 0x20, 0x4d, 0x50, 0xcb,
+    0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x32, 0xc2,
+    0x4d, 0x50, 0xce, 0x41, 0x41, 0x41, 0x41, 0x30, 0x30, 0x30, 0x30, 0x30,
+    0x30, 0x30, 0x30, 0x31, 0x32, 0xc4, 0x51, 0x45, 0x4d, 0x55, 0x00, 0xc1,
+    0x00, 0x00, 0x00, 0x86,
+};
+
+/*
+ * FRU: i2c4 @0x54 (sled 5 interposer)
+ *   Board mfg:       Quanta
+ *   Board product:   Bletchley Interposer-Class 1
+ *   Board part:      0AA00AA0001
+ *   Board serial:    AAA0A00000015
+ *   Board mfg date:  2026-01-01T00:00:00Z
+ *   Board file ID:   FRU Ver 0.04
+ *   Product mfg:     Quanta
+ *   Product name:    Bletchley MP
+ *   Product part:    0AA00AA00A2
+ *   Product version: MP
+ *   Product serial:  AAAA0000000012
+ *   Product asset:   QEMU
+ */
+static const uint8_t fru_v1_5_4_0054[] = {
+    0x01, 0x00, 0x00, 0x01, 0x0c, 0x00, 0x00, 0xf2, 0x01, 0x0b, 0x19, 0xc0,
+    0xc6, 0xf0, 0xc6, 0x51, 0x75, 0x61, 0x6e, 0x74, 0x61, 0xdc, 0x42, 0x6c,
+    0x65, 0x74, 0x63, 0x68, 0x6c, 0x65, 0x79, 0x20, 0x49, 0x6e, 0x74, 0x65,
+    0x72, 0x70, 0x6f, 0x73, 0x65, 0x72, 0x2d, 0x43, 0x6c, 0x61, 0x73, 0x73,
+    0x20, 0x31, 0xcd, 0x41, 0x41, 0x41, 0x30, 0x41, 0x30, 0x30, 0x30, 0x30,
+    0x30, 0x30, 0x31, 0x35, 0xcb, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41,
+    0x30, 0x30, 0x30, 0x31, 0xcc, 0x46, 0x52, 0x55, 0x20, 0x56, 0x65, 0x72,
+    0x20, 0x30, 0x2e, 0x30, 0x34, 0xc1, 0x00, 0x00, 0x00, 0x00, 0x00, 0xae,
+    0x01, 0x08, 0x19, 0xc6, 0x51, 0x75, 0x61, 0x6e, 0x74, 0x61, 0xcc, 0x42,
+    0x6c, 0x65, 0x74, 0x63, 0x68, 0x6c, 0x65, 0x79, 0x20, 0x4d, 0x50, 0xcb,
+    0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x32, 0xc2,
+    0x4d, 0x50, 0xce, 0x41, 0x41, 0x41, 0x41, 0x30, 0x30, 0x30, 0x30, 0x30,
+    0x30, 0x30, 0x30, 0x31, 0x32, 0xc4, 0x51, 0x45, 0x4d, 0x55, 0x00, 0xc1,
+    0x00, 0x00, 0x00, 0x86,
+};
+
+/*
+ * FRU: i2c5 @0x54 (sled 6 interposer)
+ *   (sled absent from the source chassis, serial reissued)
+ *   Board mfg:       Quanta
+ *   Board product:   Bletchley Interposer-Class 1
+ *   Board part:      0AA00AA0001
+ *   Board serial:    AAA0A00000020
+ *   Board mfg date:  2026-01-01T00:00:00Z
+ *   Board file ID:   FRU Ver 0.04
+ *   Product mfg:     Quanta
+ *   Product name:    Bletchley MP
+ *   Product part:    0AA00AA00A2
+ *   Product version: MP
+ *   Product serial:  AAAA0000000012
+ *   Product asset:   QEMU
+ */
+static const uint8_t fru_v1_5_5_0054[] = {
+    0x01, 0x00, 0x00, 0x01, 0x0c, 0x00, 0x00, 0xf2, 0x01, 0x0b, 0x19, 0xc0,
+    0xc6, 0xf0, 0xc6, 0x51, 0x75, 0x61, 0x6e, 0x74, 0x61, 0xdc, 0x42, 0x6c,
+    0x65, 0x74, 0x63, 0x68, 0x6c, 0x65, 0x79, 0x20, 0x49, 0x6e, 0x74, 0x65,
+    0x72, 0x70, 0x6f, 0x73, 0x65, 0x72, 0x2d, 0x43, 0x6c, 0x61, 0x73, 0x73,
+    0x20, 0x31, 0xcd, 0x41, 0x41, 0x41, 0x30, 0x41, 0x30, 0x30, 0x30, 0x30,
+    0x30, 0x30, 0x32, 0x30, 0xcb, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41,
+    0x30, 0x30, 0x30, 0x31, 0xcc, 0x46, 0x52, 0x55, 0x20, 0x56, 0x65, 0x72,
+    0x20, 0x30, 0x2e, 0x30, 0x34, 0xc1, 0x00, 0x00, 0x00, 0x00, 0x00, 0xb2,
+    0x01, 0x08, 0x19, 0xc6, 0x51, 0x75, 0x61, 0x6e, 0x74, 0x61, 0xcc, 0x42,
+    0x6c, 0x65, 0x74, 0x63, 0x68, 0x6c, 0x65, 0x79, 0x20, 0x4d, 0x50, 0xcb,
+    0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x32, 0xc2,
+    0x4d, 0x50, 0xce, 0x41, 0x41, 0x41, 0x41, 0x30, 0x30, 0x30, 0x30, 0x30,
+    0x30, 0x30, 0x30, 0x31, 0x32, 0xc4, 0x51, 0x45, 0x4d, 0x55, 0x00, 0xc1,
+    0x00, 0x00, 0x00, 0x86,
+};
+
+/*
+ * FRU: i2c6 @0x56 (BMC storage module)
+ *   Chassis type:    Rack Mount Chassis
+ *   Board mfg:       Quanta
+ *   Board product:   BMC Storage Module
+ *   Board part:      0AA0AAA00A4
+ *   Board serial:    AAA0A00000016
+ *   Board mfg date:  2026-01-01T00:00:00Z
+ *   Board file ID:   FRU Version 0.02
+ *   Product mfg:     Quanta
+ *   Product name:    Bletchley v1.5 MP
+ *   Product part:    0AA00AA00A2
+ *   Product version: MP
+ *   Product serial:  AAAA0000000012
+ *   Product asset:   QEMU
+ */
+static const uint8_t fru_v1_5_6_0056[] = {
+    0x01, 0x00, 0x01, 0x02, 0x0c, 0x00, 0x00, 0xf0, 0x01, 0x01, 0x17, 0x00,
+    0x00, 0xc1, 0x00, 0x26, 0x01, 0x0a, 0x19, 0xc0, 0xc6, 0xf0, 0xc6, 0x51,
+    0x75, 0x61, 0x6e, 0x74, 0x61, 0xd2, 0x42, 0x4d, 0x43, 0x20, 0x53, 0x74,
+    0x6f, 0x72, 0x61, 0x67, 0x65, 0x20, 0x4d, 0x6f, 0x64, 0x75, 0x6c, 0x65,
+    0xcd, 0x41, 0x41, 0x41, 0x30, 0x41, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30,
+    0x31, 0x36, 0xcb, 0x30, 0x41, 0x41, 0x30, 0x41, 0x41, 0x41, 0x30, 0x30,
+    0x41, 0x34, 0xd0, 0x46, 0x52, 0x55, 0x20, 0x56, 0x65, 0x72, 0x73, 0x69,
+    0x6f, 0x6e, 0x20, 0x30, 0x2e, 0x30, 0x32, 0xc1, 0x00, 0x00, 0x00, 0xe6,
+    0x01, 0x09, 0x19, 0xc6, 0x51, 0x75, 0x61, 0x6e, 0x74, 0x61, 0xd1, 0x42,
+    0x6c, 0x65, 0x74, 0x63, 0x68, 0x6c, 0x65, 0x79, 0x20, 0x76, 0x31, 0x2e,
+    0x35, 0x20, 0x4d, 0x50, 0xcb, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41,
+    0x30, 0x30, 0x41, 0x32, 0xc2, 0x4d, 0x50, 0xce, 0x41, 0x41, 0x41, 0x41,
+    0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x31, 0x32, 0xc4, 0x51,
+    0x45, 0x4d, 0x55, 0x00, 0xc1, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x56,
+};
+
+/*
+ * FRU: i2c7 @0x54 (chassis controller)
+ *   Chassis type:    Rack Mount Chassis
+ *   Chassis part:    0AA00AA0002
+ *   Chassis serial:  AAA0AA00AA00000AA0A017A
+ *   Chassis custom:  9082C3000002
+ *   Board mfg:       Quanta
+ *   Board product:   Bletchley v1.5 Chassis Controller-Class 1
+ *   Board part:      0AA00AA0005
+ *   Board serial:    AAA0A00000018
+ *   Board mfg date:  2026-01-01T00:00:00Z
+ *   Board file ID:   FRU Version 0.02
+ *   Product mfg:     Quanta
+ *   Product name:    Bletchley v1.5 MP
+ *   Product part:    0AA00AA00A2
+ *   Product version: MP
+ *   Product serial:  AAAA0000000012
+ *   Product asset:   QEMU
+ */
+static const uint8_t fru_v1_5_7_0054[] = {
+    0x01, 0x00, 0x01, 0x08, 0x15, 0x00, 0x00, 0xe1, 0x01, 0x07, 0x17, 0xcb,
+    0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41, 0x30, 0x30, 0x30, 0x32, 0xd7,
+    0x41, 0x41, 0x41, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41, 0x30, 0x30,
+    0x30, 0x30, 0x30, 0x41, 0x41, 0x30, 0x41, 0x30, 0x31, 0x37, 0x41, 0xc0,
+    0xcc, 0x39, 0x30, 0x38, 0x32, 0x43, 0x33, 0x30, 0x30, 0x30, 0x30, 0x30,
+    0x32, 0xc1, 0x00, 0x1e, 0x01, 0x0d, 0x19, 0xc0, 0xc6, 0xf0, 0xc6, 0x51,
+    0x75, 0x61, 0x6e, 0x74, 0x61, 0xe9, 0x42, 0x6c, 0x65, 0x74, 0x63, 0x68,
+    0x6c, 0x65, 0x79, 0x20, 0x76, 0x31, 0x2e, 0x35, 0x20, 0x43, 0x68, 0x61,
+    0x73, 0x73, 0x69, 0x73, 0x20, 0x43, 0x6f, 0x6e, 0x74, 0x72, 0x6f, 0x6c,
+    0x6c, 0x65, 0x72, 0x2d, 0x43, 0x6c, 0x61, 0x73, 0x73, 0x20, 0x31, 0xcd,
+    0x41, 0x41, 0x41, 0x30, 0x41, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x31,
+    0x38, 0xcb, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41, 0x30, 0x30, 0x30,
+    0x35, 0xd0, 0x46, 0x52, 0x55, 0x20, 0x56, 0x65, 0x72, 0x73, 0x69, 0x6f,
+    0x6e, 0x20, 0x30, 0x2e, 0x30, 0x32, 0xc1, 0x00, 0x00, 0x00, 0x00, 0xcc,
+    0x01, 0x09, 0x19, 0xc6, 0x51, 0x75, 0x61, 0x6e, 0x74, 0x61, 0xd1, 0x42,
+    0x6c, 0x65, 0x74, 0x63, 0x68, 0x6c, 0x65, 0x79, 0x20, 0x76, 0x31, 0x2e,
+    0x35, 0x20, 0x4d, 0x50, 0xcb, 0x30, 0x41, 0x41, 0x30, 0x30, 0x41, 0x41,
+    0x30, 0x30, 0x41, 0x32, 0xc2, 0x4d, 0x50, 0xce, 0x41, 0x41, 0x41, 0x41,
+    0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x31, 0x32, 0xc4, 0x51,
+    0x45, 0x4d, 0x55, 0x00, 0xc1, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x56,
+};
+
+static const BletchleyFruSet bletchley_fru_v1_5 = {
+    .sled = {
+        { fru_v1_5_0_0054, sizeof(fru_v1_5_0_0054) },
+        { fru_v1_5_1_0054, sizeof(fru_v1_5_1_0054) },
+        { fru_v1_5_2_0054, sizeof(fru_v1_5_2_0054) },
+        { fru_v1_5_3_0054, sizeof(fru_v1_5_3_0054) },
+        { fru_v1_5_4_0054, sizeof(fru_v1_5_4_0054) },
+        { fru_v1_5_5_0054, sizeof(fru_v1_5_5_0054) },
+    },
+    .bsm = { fru_v1_5_6_0056, sizeof(fru_v1_5_6_0056) },
+    .cc  = { fru_v1_5_7_0054, sizeof(fru_v1_5_7_0054) },
+};
+
 #define TYPE_BLETCHLEY_MACHINE MACHINE_TYPE_NAME("bletchley-bmc")
 OBJECT_DECLARE_SIMPLE_TYPE(BletchleyMachineState, BLETCHLEY_MACHINE)
 
@@ -343,7 +627,8 @@ struct BletchleyMachineState {
 
 static void bletchley_bmc_i2c_init(AspeedMachineState *bmc)
 {
-    const BletchleyFruSet *fru = &bletchley_fru_v1_0;
+    const BletchleyFruSet *fru = BLETCHLEY_MACHINE(bmc)->v1_5 ?
+                                 &bletchley_fru_v1_5 : &bletchley_fru_v1_0;
     AspeedSoCState *soc = bmc->soc;
     I2CBus *i2c[13] = {};
     for (int i = 0; i < 13; i++) {
-- 
2.53.0-Meta

